docs/en/examples/elementalarea_with_dataobject.md

```diff
@@ -0,0 +1,161 @@
+    ];
+}
+```
+
+`$managed_models` - this static variable connects BlogPostAdmin with BlogPost, which will be created in the following code.
+When you create a DataObject you should define the connection between your DataObject and ElementalArea by adding `$has_one` relationship.
+In this example BlogPost will have one ElementalArea.
+
+**app/src/Models/BlogPost.php**
+
+```php
+<?php
+
+namespace App\Models;
+
+use App\Admins\BlogPostsAdmin;
+use DNADesign\Elemental\Models\ElementalArea;
+use SilverStripe\Control\Controller;
+use SilverStripe\ORM\DataObject;
+use SilverStripe\Versioned\Versioned;
+
+class BlogPost extends DataObject
+{
+    private static string $table_name = 'BlogPost';
+
+    private static $extensions = [
+        Versioned::class,
+    ];
+
+    private static array $db = [
+        'Title' => 'Varchar(255)',
+        'Content' => 'HTMLText',
+    ];
+
+    private static $has_one = [
+        'ElementalArea' => ElementalArea::class,
+    ];
+
+    private static $owns = ['ElementalArea'];
+
+    // ...
+}
+ ```
+
+If you are using ElementalArea together with DataObject, it is important to define a few following methods in the class.
+First of all, the CMSEditLink() method should be defined to generate a link for opening the editing section of the particular BlogPost.
+BaseElement uses a link, that was created in CMSEditLink() method in BlogPost class, to generate new links for opening the editing section of the particular Element.
+
+```php
+// ...
+
+class BlogPost extends DataObject
+{
+    // ...
+
+    public function CMSEditLink()
+    {
+        // In this example we use BlogPostsAdmin class as Controller
+        $admin = BlogPostsAdmin::singleton();
+
+        // Makes link more readable. Instead App\Models\ BlogPost, we get App-Models-BlogPost
+        $sanitisedClassname = str_replace('\\', '-', $this->ClassName);
+
+        // Returns link to editing section with elements
+        return Controller::join_links(
+            $admin->Link($sanitisedClassname),
+            'EditForm/field/',
+            $sanitisedClassname,
+            'item',
+            $this->ID,
+        );
+    }
+
+    // ...
+}
+
+```
+
+Method Link() doesn't exist in DataObject class, so you should define this method in your class. You should add the canArchive() method in your class as well to set rules for the canDelete() method.
+You should also define the AbsoluteLink() method for opening the tabs in the Elements.
+Use the following code as an example of how to simply create these methods.
+
+```php
+// ...
+
+class BlogPost extends DataObject
+{
+    // ...
+
+    public function Link($action = null)
+    {
+        $admin = BlogPostsAdmin::singleton();
+
+        return $this->ID
+            ? Controller::join_links(
+                $admin->Link(str_replace('\\', '-', $this->ClassName)),
+                $this->ID
+                )
+            : null;
+    }
+
+    public function canArchive($member = null)
```

Comment:
Why are we explicitly mentioning `canArchive` here? The class doesn't have to be `Versioned`.

Reply:
Method canDelete() in BaseElement class invoke method canArchive on $page. Probably, it's expected that page is Versioned. And from Steve's comment https://github.com/silverstripe/silverstripe-elemental/pull/978#pullrequestreview-955619575, it's expected that possibly DataObject instance can be Versioned as well.

Reply:
It would be nicer if you add a check in `BaseElement` - something like
```php
if ($page->hasExtension(Versioned::class)) {
    // use $page->canArchive()
} else {
    // use $page->canDelete()
}
```